### PR TITLE
Feature/2.4.4 compatibility

### PR DESCRIPTION
**Pull request by supercid**
## Description
- Update PHP SDK
- Bump PHP min version to 7.4
- Suppress PHP 8.1 inheritance warning (since 7.4 does not have typehint return for mixed, which is needed)
- Little bit more strict types
- Better types in general

## Related Issue
Not public

## Motivation and Context
Magento 2.4.4 came with significant changes to how we handle the data schema, requiring removal of PHP based schema to XML.

## How Has This Been Tested?
Tested locally with a brand new docker image for Magento 2.4.4 and PHP 8.1
You can test by using https://github.com/supercid/magento-docker-bin
- Clone the repo
- Run `run_first_time.sh` [run_first_time.sh](https://github.com/supercid/magento-docker-bin/blob/main/run_first_time.sh)
- Feel free to test locally at https://magento2.dev.nos.to

Tested in plugintest at https://magento244.plugintest.nos.to/

## Documentation:
No updates needed

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Console/Command/NostoAccountConnectCommand.php [hunk 1]
```diff
@@ -213,7 +213,7 @@ private function updateNostoTokens(array $tokens, $accountId, SymfonyStyle $io,
      */
     private function getTokensFromInput(InputInterface $input, SymfonyStyle $io)
     {
```
Comment (dairbuirabass) [inline Console/Command/NostoAccountConnectCommand.php:215]: Is there a way to get rid of these warning? 🤔 @supercid

File: Controller/Adminhtml/Account/Proxy.php [hunk 2]
```diff
@@ -56,6 +59,7 @@ public function __construct(
     ) {
         parent::__construct($context);
 
+        $this->_publicActions = ['proxy'];
```
Comment (dairbuirabass) [inline Controller/Adminhtml/Account/Proxy.php:62]: Would it make sense to move into a constant?

File: compile.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@ PROJECT_NAME=$(cat composer.json| jq --raw-output .name)
 APP_DIR=$(pwd)
 TMP_DIR=$(mktemp -d -t ci-XXXXXXXXXX)
 cd $TMP_DIR || exit 0
-composer create-project magento/community-edition=2.3.2 .
+composer create-project magento/community-edition=2.4.3 .
```
Comment (dairbuirabass) [inline compile.sh:7]: Should be 2.4.4 now?

File: Controller/Adminhtml/Account/Proxy.php (hunk 2 above)
Comment (supercid) [reply Controller/Adminhtml/Account/Proxy.php:62]: Comes from a Magento base class, better not to touch it :)

File: Helper/Price.php [hunk 4]
```diff
@@ -134,9 +134,9 @@ public function getProductDisplayPrice(Product $product, Store $store)
      */
     public function getProductPrice(// @codingStandardsIgnoreLine
         Product $product,
-        Store $store,
-        $inclTax = true,
-        $finalPrice = false
+        Store   $store,
+        bool    $inclTax = true,
+        bool $finalPrice = false
```
[the remark this answers is not in the document]
Comment (supercid) [reply Helper/Price.php:139]: Fixed

File: compile.sh (hunk 3 above)
Comment (supercid) [reply compile.sh:7]: Fixed

File: Console/Command/NostoAccountConnectCommand.php (hunk 1 above)
Comment (supercid) [reply Console/Command/NostoAccountConnectCommand.php:215]: Yeah, we can configure the sniffer later on, this PR is already too long

**Review by dairbuirabass — APPROVED** [review]

**Pull request merged**